Feature/translate en (#7)
stack-spot/mind-dotnet-worker-stack: docs/use-case_en-us.md

- Install [**`STK CLI`**](https://stackspot.com/download)
- **.NET 6**.

### **Inputs**
The inputs needed to use the template are: 

| **Field** | **Value** | **Description** |
| :--- | :--- | :--- |
| Target Framework| .net6 | Target Framework in which the application will be created  |

## Running the project

After creating the project, access the directory where it was created and run the following command:

- Replace the `*` with the given name for application.

```bash
dotnet restore *.Worker.sln
```

Also build the project, using the command below:

```bash
dotnet build *.Worker.sln
```

In order to run the unit and integration tests use the command below:

```bash
dotnet test *.Worker.sln
```

To run the application, in the same directory, use the following command:

```bash
dotnet run --project ./src/*.Worker/*.Worker.csproj
```

### Docker Configurations

To build and run docker containers, run the command below in the solution root where you find the docker-compose.yml file.

 ```bash
 docker-compose -f 'docker-compose.yml' up --build
 ```

> You can also use a **IDE** (VSCode, Visual Studio) of your choice to perform the above steps.

### Base Template Capabilities

> Check below the capabilities available to be added to a base Template.

Reviewer: ```suggestion
Check the capabilities available that you can add to a base Template.
```

- **Logging**  

> If you have several projects, microservices and/or routines, with this Plugin you can maintain the quality and standard of the log information for everything, considering the different log levels.
> 
> It's allow you to define Tags so that you can quantify and/or classify errors, providing a view of recurring failures or preventing future problems.

Reviewer: ```suggestion
If you have several projects, microservices, and/or routines, with this Plugin you can maintain the quality and standard of the log information for everything, considering different log levels.

It allows you to define Tags, and you can quantify and/or classify errors, providing a view of recurring failures or preventing future problems.
```

> In addition, it is easy to integrate with tools such as **Splunk** or **AWS Cloud Watch** to monitor and visualize data, ensuring the observability of your applications.

Reviewer: ```suggestion
It is also easy to integrate with tools such as **Splunk** or **AWS Cloud Watch** to monitor and visualize data, ensuring the observability of your applications.
```

- **Metrics**    
The Metrics Plugin makes it easy to measure the number of requests on a critical endpoint or monitor the size of payloads that are sent to your API.

- **Trace**    

> Trace Plugin facilitates telemetry instrumentation of your API, through **Opentelemetry**, with exporter for **Jaeger** and **AWS X-Ray**.

Reviewer: ```suggestion
Trace Plugin facilitates telemetry instrumentation of your API, through **Opentelemetry**, with an exporter for **Jaeger** and **AWS X-Ray**.
```

- **Queue**    

> The Queue Plugin simplifies the entire configuration and implementation of a decentralized environment, focused on microservices that communicate with each other, using message queues. Compared to creating a class, Plugin Queue manages to abstract all the complexity for communicating with your **AWS SQS** queue.

Reviewer: ```suggestion
The Queue Plugin simplifies the entire configuration and implementation of a decentralized environment, focused on microservices that communicate with each other, using message queues. It's like creating a class, the Plugin Queue manages to abstract all the complexity communicating with your **AWS SQS** queue.
```

- **Notification**    

> The Notification Plugin provides the features of **AWS SNS (Simple Notification Service)** so that your application can send messages to topics that will be consumed by your Subscribers, simultaneously sending data, information and messages to different platforms at once.

Reviewer: ```suggestion
The Notification Plugin provides the **AWS SNS (Simple Notification Service)** features so your application can send messages to topics that will be consumed by your Subscribers, simultaneously sending data, information, and messages to different platforms at once.
```

- **Secrets**    

> Plugin Secrets turns easier the management of sensitive system information with API keys, authentication tokens or database connections. It was thought of using a **Secret Manager** from AWS simplifying the consumption of this information.
> Use the Secrets plugin any time you want within your application.

Reviewer: ```suggestion
Plugin Secrets makes the management of sensitive system information with API keys, authentication tokens, or database connections easier. It was thought of using a **Secret Manager** from AWS to simplify the consumption of this information.
Use the Secrets plugin any time you want in your application.
```

- **Bucket**    

> Handle files using Plugin Bucket, it eliminates complexity and facilitates instrumentation with **Amazon Simple Storage Service (S3)**.

Reviewer: ```suggestion
You can handle files using the Bucket Plugin. It eliminates the complexity and facilitates instrumentation with **Amazon Simple Storage Service (S3)**.
```

- **Repository**    

> Store data or documents trivially. Plugin Repository brings together all the capabilities of **AWS DynamoDB** in a simple and fast way.

Reviewer: ```suggestion
Store data or documents trivially. Repository Plugin brings together all the capabilities of **AWS DynamoDB** in a simple and fast way.
```

- **Authentication**    

> Plugin Authentication provides more security for you to easily verify a user/API accesses to the restricted resources of your API.

Reviewer: ```suggestion
Authentication Plugin provides more security for you to easily verify a user/API access to the restricted resources of your API.
```

- **ErrorHandler**  
Handle, standardize and customize error returns in your REST API.

- **gRPC**  

> The gRPC Plugin adds the ability to expose `gRPC` services and their settings, and provide the creation of a `gRPC Client`, that allows the distribution and consumption of this service in other applications.

Reviewer: ```suggestion
The gRPC Plugin adds the ability to expose `gRPC` services and their settings. It provides the creation of a `gRPC Client`, that allows the distribution and consumption of this service in other applications.
```

> **Check out more details such as ways to use and use cases of each Plugin in the left side menu.**

Reviewer: ```suggestion
**See more details like ways to use and use cases of each Plugin on the left side menu.**
```
